header.php

```diff
@@ -23,6 +23,31 @@
 	<!-- wordpress head functions -->
 	<?php wp_head(); ?>
 
+  <link rel="apple-touch-icon" sizes="180x180" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/apple-touch-icon.png">
+  <link rel="icon" type="image/png" sizes="32x32" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/favicon-32x32.png">
+  <link rel="icon" type="image/png" sizes="16x16" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/favicon-16x16.png">
+  <link rel="manifest" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/site.webmanifest">
+  <link rel="mask-icon" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/safari-pinned-tab.svg" color="#4881cb">
+  <link rel="shortcut icon" href="<?php echo get_stylesheet_directory_uri() ?>/library/icons/favicon.ico">
+  <meta name="apple-mobile-web-app-title" content="PPSri.org">
+  <meta name="application-name" content="PPSri.org">
+  <meta name="msapplication-TileColor" content="#4881cb">
+  <meta name="msapplication-TileImage" content="<?php echo get_stylesheet_directory_uri() ?>/library/icons/mstile-150x150.png">
+  <meta name="msapplication-TileImage" content="<?php echo get_stylesheet_directory_uri() ?>/library/icons/mstile-144x144.png">
```

Comment:
Why not use wordpress core's Site Icon? https://www.sitepoint.com/all-you-need-to-know-about-the-new-wordpress-site-icon-api/ You just upload an image once and it will create much of this for you. You can set up custom sizes for anything you want that isn't there by default.

Reply:
WordPress core favicon is incomplete. Or, it covers the very basic subset of icons needed for mobile and desktop. When I use WP core it only produces four lines of code, covering iOS and classic desktop browsers. Android, Windows, & OSX El Capitan are not covered and the checker at realfavicongenerator.net says that the `favicon/ico` file is ill-formed and missing some of the recommended embedded sizes. The tile icon that is created for Metro is just the same as the touch icon, therefore it does not adhere to the Windows Metro standards of a white transparent PNG. 

On top of all that, I have curated the different sizes needed with different artwork. As the icon gets smaller, the complicated logo artwork becomes cropped and simpler so there is still some legibility of the mark even at the 16x16 size. 

In short, I prefer to complete set of meta data rather than WP's minimum.

Reply:
Okay! Thanks for the explanation. I think there is a way to hook into the existing core functionality and expand it, but I'm not sure if you can "curate" the sizes like you're saying. Let's go with what you have.